sklearn/neighbors/tests/test_lmnn.py

```diff
@@ -0,0 +1,677 @@
+
+    assert_raise_message(ValueError,
+                         '`n_neighbors`= {}, must be <= {}.'
+                         .format(X.shape[0], X.shape[0] - 1),
+                         LMNN(n_neighbors=X.shape[0]).fit, X, y)
+
+    assert_raise_message(ValueError,
+                         '`max_iter`= -1, must be >= 1.',
+                         LMNN(max_iter=-1).fit, X, y)
+    assert_raise_message(ValueError,
+                         '`max_impostors`= -1, must be >= 1.',
+                         LMNN(max_impostors=-1).fit, X, y)
+    assert_raise_message(ValueError,
+                         "`impostor_store` must be 'auto', 'sparse' "
+                         "or 'list'.",
+                         LMNN(impostor_store='dense').fit, X, y)
+
+    assert_raise_message(ValueError,
+                         '`weight_push_loss`= 2.0, must be <= 1.0.',
+                         LMNN(weight_push_loss=2.).fit, X, y)
+
+    assert_raise_message(ValueError,
+                         '`weight_push_loss` cannot be zero.',
+                         LMNN(weight_push_loss=0.).fit, X, y)
+
+    init = np.random.rand(5, 3)
+    assert_raise_message(ValueError,
+                         'The output dimensionality ({}) of the given linear '
+                         'transformation `init` cannot be greater than its '
+                         'input dimensionality ({}).'
+                         .format(init.shape[0], init.shape[1]),
+                         LMNN(init=init).fit, X, y)
+
+    n_components = 10
+    assert_raise_message(ValueError,
+                         'The preferred embedding dimensionality '
+                         '`n_components` ({}) cannot be greater '
+                         'than the given data dimensionality ({})!'
+                         .format(n_components, X.shape[1]),
+                         LMNN(n_components=n_components).fit, X, y)
+
+    n_jobs = 0
+    assert_raise_message(ValueError,
+                         'n_jobs == 0 in Parallel has no meaning',
+                         LMNN(n_jobs=n_jobs).fit, X, y)
+
+    # test min_class_size < 2
+    y = [1, 1, 1, 2]
+    assert_raise_message(ValueError,
+                         'LargeMarginNearestNeighbor needs at least 2 '
+                         'non-singleton classes, got 1.',
+                         LMNN(n_neighbors=1).fit, X, y)
+
+
+def test_n_neighbors():
+    X = np.arange(12).reshape(4, 3)
+    y = [1, 1, 2, 2]
+
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=2)
+    assert_warns_message(UserWarning,
+                         '`n_neighbors` (=2) is not less than the number of '
+                         'samples in the smallest non-singleton class (=2). '
+                         '`n_neighbors_` will be set to 1 for estimation.',
+                         lmnn.fit, X, y)
+
+
+def test_n_components():
+    X = np.arange(12).reshape(4, 3)
+    y = [1, 1, 2, 2]
+
+    init = np.random.rand(X.shape[1] - 1, 3)
```

Comment:
Maybe use a local random seed for the test to be deterministic ?

Reply:
Done.